conformance/tests/generics_paramspec_semantics.py

```diff
@@ -0,0 +1,143 @@
+"""
+Tests semantics of ParamSpec.
+"""
+
+# Specification: https://typing.readthedocs.io/en/latest/spec/generics.html#semantics
+
+
+from typing import Callable, Concatenate, Generic, ParamSpec, TypeVar, assert_type
+
+P = ParamSpec("P")
+
+
+def changes_return_type_to_str(x: Callable[P, int]) -> Callable[P, str]:
+    ...
+
+
+def returns_int(a: str, b: bool, /) -> int:
+    ...
+
+
+f1 = changes_return_type_to_str(returns_int)
+assert_type(f1, Callable[[str, bool], str])
+
+v1 = f1("A", True)  # OK
+assert_type(v1, str)
+f1(a="A", b=True)  # Type error: position-only
+f1("A", "A")  # Type error: wrong type
+
+
+def func1(x: Callable[P, int], y: Callable[P, int]) -> Callable[P, bool]:
+    ...
+
+
+def x_y(x: int, y: str) -> int:
+    ...
+
+
+def y_x(y: int, x: str) -> int:
+    ...
+
+
+f2 = func1(x_y, x_y)
+assert_type(f2(1, ""), bool)
+assert_type(f2(y="", x=1), bool)
+
+f3 = func1(x_y, y_x)  # Could return (a: int, b: str, /) -> bool
+# (a callable with two positional-only parameters)
+# This works because both callables have types that are
+# behavioral subtypes of Callable[[int, str], int]
+# Also OK for type checkers to reject this.
+
+
+def keyword_only_x(*, x: int) -> int:
+    ...
+
+
+def keyword_only_y(*, y: int) -> int:
+    ...
+
+
+func1(keyword_only_x, keyword_only_y)  # Type error
+
+
+U = TypeVar("U")
+
+
+class Y(Generic[U, P]):
+    f: Callable[P, str]
+    prop: U
+
+    def __init__(self, f: Callable[P, str], prop: U) -> None:
+        self.f = f
+        self.prop = prop
+
+
+def callback_a(q: int, /) -> str:
+    ...
+
+
+y1 = Y(callback_a, 1)
+assert_type(y1, Y[int, [int]])
```

Comment:
Spec says `# Should resolve to Y[(q: int), int]` which I think has the wrong ordering

Reply:
I agree that's a typo in the spec. I included a fix.